src/lib/core/compatibility/compatibility.ts

```diff
@@ -193,11 +194,16 @@ export class CompatibilityModule {
     };
   }
 
-  constructor(@Optional() @Inject(DOCUMENT) private _document: any) {
+  constructor(@Optional() @Inject(DOCUMENT) private _document: any, ngZone: NgZone) {
     if (!hasDoneGlobalChecks && isDevMode()) {
-      this._checkDoctype();
-      this._checkTheme();
-      hasDoneGlobalChecks = true;
+      ngZone.runOutsideAngular(() => {
```

Comment:
Can we gate on being on the browser platform here rather than in the individual checks as well?
(no reason the timeout has to run on platform-server)